capstone/src/main/java/com/google/sps/servlets/AnnouncementsServlet.java

```diff
@@ -12,18 +25,53 @@
 @WebServlet("/announcements")
 public class AnnouncementsServlet extends HttpServlet {
 
+  private static final String CLUB_NAME_PROP = "name";
+  private static final String ANNOUNCEMENTS_CONTENT_PROP = "content";
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get announcements object based on the logged in email
+    UserService userService = UserServiceFactory.getUserService();
+    String userEmail = userService.getCurrentUser().getEmail();
+    String clubName = request.getParameter(CLUB_NAME_PROP);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    Query query =
+        new Query("Announcement")
+            .setFilter(new FilterPredicate("club", FilterOperator.EQUAL, clubName))
+            .addSort("time", SortDirection.DESCENDING);
+    PreparedQuery results = datastore.prepare(query);
+    ImmutableList<Announcement> announcements =
+        Streams.stream(results.asIterable())
+            .limit(Constants.LOAD_LIMIT)
+            .map(entity -> new Announcement(entity))
+            .collect(toImmutableList());
+
     Gson gson = new Gson();
-    ImmutableList<String> announcements =
-        PrototypeClubs.PROTOTYPE_CLUBS_MAP
-            .get(request.getParameter(Constants.CLUB_NAME_PROP))
-            .getAnnouncements();
     String json = gson.toJson(announcements);
     response.setContentType("application/json;");
     response.getWriter().println(json);
   }
 
   @Override
-  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {}
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    UserService userService = UserServiceFactory.getUserService();
+    String userEmail = userService.getCurrentUser().getEmail();
+    String clubName = request.getParameter(CLUB_NAME_PROP);
+    String announcementContent = request.getParameter(ANNOUNCEMENTS_CONTENT_PROP);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    // TODO need to authenticate user as club officer
+
+    Entity announcementEntity = new Entity("Announcement");
+    announcementEntity.setProperty("author", userEmail);
```

Comment:
Might be good to make "author" and the other property names constants, at the top of the file or something (you just reminded me that I should do this too lol). Then maybe if there are overlapping constnats we can put it in Constants.java